### ISSUE TITLE
MarkerLoggingAdapter's error method eating cause parameters

### ISSUE THREAD

**Issue by sumnulu** [akka/akka#22324]
akka.event.MarkerLoggingAdapter does not using cause parameter of the error methods.
akka version = 2.4.16

```SCALA
// ....................
/**
 * [[LoggingAdapter]] extension which adds Marker support.
 * Only recommended to be used within Actors as it isn't thread safe.
 */
class MarkerLoggingAdapter(
  override val bus:       LoggingBus,
  override val logSource: String,
  override val logClass:  Class[_],
  loggingFilter:          LoggingFilter)
  extends BusLogging(bus, logSource, logClass, loggingFilter) {
  // TODO when breaking binary compatibility, these marker methods should become baked into LoggingAdapter itself

  // For backwards compatibility, and when LoggingAdapter is created without direct
  // association to an ActorSystem
  def this(bus: LoggingBus, logSource: String, logClass: Class[_]) =
    this(bus, logSource, logClass, new DefaultLoggingFilter(() ⇒ bus.logLevel))

  /**
   * Log message at error level, including the exception that caused the error.
   * The marker argument can be picked up by various logging frameworks such as slf4j to mark this log statement as "special".
   * @see [[LoggingAdapter]]
   */
  def error(marker: LogMarker, cause: Throwable, message: String): Unit =
    if (isErrorEnabled) bus.publish(Error(logSource, logClass, message, mdc, marker))
// ...........
```

example :
```SCALA
  val s: DiagnosticMarkerBusLoggingAdapter = Logging.withMarker(this)
  s.error(LogMarker("TEST_MARKER"), new UnsupportedOperationException("test"), "contains exception error")
```

**Comment by johanandren** [comment]
Did you configure your logger implementation to include the markers in its output (described in the docs for SLF4J here: http://doc.akka.io/docs/akka/2.4.17/scala/logging.html#Using_Markers)?

**Comment by sumnulu** [comment]
Yes I have. It is logging the markers but not the exception. If I use ActorLogging logback is logging the exception without any problem.

The problem is I think the MarkerLoggingAdapter does not using the cause parameter in the error method's body and just ignoring it. (see my previous comment's first snippet (it is the source code of akka) )

My logsback.xml pattern is like this one:
`<pattern>%date{HH:mm:ss} %highlight(%-5level) %boldYellow(%marker) %magenta(%logger{30}) %boldGreen(%X{peerId}) - %msg %ex%n</pattern>`

**Comment by patriknw** [comment]
@sumnulu I agree, this is an oversight. It doesn't pass on the `throwable` parameter. Would you be able to create a Pull Request fixing this issue?

**Comment by sumnulu** [comment]
@patriknw ok, I will.

**Comment by patriknw** [comment]
thanks

**Comment by jrudolph** [comment]
That is solved as part as #23109.

**Comment by patriknw** [comment]
@raboof  I think this is serious enough to warrant a backport to 2.4 (and don't forget to set the milestone)
